src/lib/tls/credentials_manager.h

```diff
@@ -93,24 +93,23 @@ class BOTAN_PUBLIC_API(2, 0) Credentials_Manager {
                                                               const std::string& context);
 
       /**
-      * Return a cert chain we can use, ordered from leaf to root,
-      * or else an empty vector.
+      * Return a certificate chain we can use to identify ourselves, ordered
+      * from leaf to root, or else an empty vector.
       *
-      * This virtual function is deprecated, and will be removed in a
-      * future release. Use (and override) find_cert_chain instead.
+      * @deprecated This virtual function is deprecated, and will be removed in
+      *             a future release. Use (and override) find_cert_chain()
+      *             instead.
```

Comment:
Clang warns that the docstring claims a deprecation but there's no `[[deprecated]]` attribute in the function's declaration.
This would be fixable by adding `BOTAN_DEPRECATED()`. However, when we build with the deprecation annotations being disabled (e.g. amalgamation builds), this clang warning might still show up and break things (due to `-Werror`).

I think we should either disable this clang warning (`-Wdocumentation-deprecated-sync`), or not use `@deprecated` in the Doxygen comments. This seems to be the only/first location where we use it anyway.

Reply:
That said: clang has a point. We should actually mark this as deprecated in favor of `find_cert_chain()`. When invoked in a client-auth context, `find_cert_chain()` conveniently delivers information about the advertised acceptable CAs of the server.

Reply:
I removed the `@deprecated` comment but also did introduce a `BOTAN_DEPRECATED` annotation here.